### PR TITLE
Metadata filter sample ids by taxonomy and alpha

### PR DESCRIPTION
**Pull request by gwarmstrong**
Adds a feature for filtering the sample ids in the metadata by a given table with taxonomy and/or alpha metric data.

### CODE REVIEW

File: microsetta_public_api/api/tests/test_integration.py [hunk 1]
```diff
@@ -201,6 +221,51 @@ def test_group_summary(self):
 class AllIntegrationTest(
         AlphaIntegrationTests,
         TaxonomyIntegrationTests,
+        MetadataIntegrationTests,
         ):
 
-    pass
+    def test_metadata_filter_on_taxonomy(self):
+        response = self.client.get('/api/metadata/sample-ids?taxonomy=table2')
+        print(response.data)
```
Comment (wasade) [inline microsetta_public_api/api/tests/test_integration.py:229]: left over print

File: microsetta_public_api/api/tests/test_integration.py [hunk 2]
```diff
@@ -201,6 +221,51 @@ def test_group_summary(self):
...
+        self.assertEqual(response.status_code, 200)
+        obs = json.loads(response.data)
+        self.assertCountEqual(['sample-1', 'sample-2', 'sample-3'],
+                              obs['sample_ids'])
+
+    def test_metadata_filter_on_taxonomy_and_age_cat(self):
+        response = self.client.get(
+            '/api/metadata/sample-ids?taxonomy=table2&age_cat=50s')
+        print(response.data)
```
Comment (wasade) [inline microsetta_public_api/api/tests/test_integration.py:238]: ...seems to be a few of these

Comment (gwarmstrong) [reply microsetta_public_api/api/tests/test_integration.py]: Can you tell which tests I broke during development 🙃 
Fixed in https://github.com/biocore/microsetta-public-api/pull/19/commits/b91aa55c753668901aa122e3bb0c4bb7907d0190

File: microsetta_public_api/api/metadata.py [hunk 3]
```diff
@@ -12,13 +14,45 @@ def category_values(category):
...
-def filter_sample_ids(**kwargs):
+def filter_sample_ids(taxonomy=None, alpha_metric=None, **kwargs):
     repo = MetadataRepo()
     query = _format_query(kwargs)
     is_invalid = _validate_query(kwargs, repo)
     if is_invalid:
         return is_invalid
     matching_ids = repo.sample_id_matches(query)
+
+    if taxonomy is not None:
```
[the remark this answers is not in the document]
Comment (gwarmstrong) [reply microsetta_public_api/api/metadata.py]: Howdya like this one? https://github.com/biocore/microsetta-public-api/pull/19/commits/ed38944a998d2b5d474cd1d71467a83196ab20f4

**Pull request merged**